Modules/rat_command.py

```diff
@@ -55,143 +53,168 @@ class NameCollisionException(CommandException):
 
-    @classmethod
-    async def trigger(cls, message: str, sender: str, channel: str):
-        """
-        Invoke a command, passing args and kwargs to the called function
-        :param message: triggers message to invoke
-        :param sender: author of triggering message
-        :param channel: channel of triggering message
-        :return: bool command
-        """
-        if cls.bot is None:
-            # someone didn't set me.
-            raise CommandException(f"Bot client has not been created"
-                                   f" or not handed to Commands.")
-
-        # check for trigger
-        assert message.startswith(cls.prefix), f"message passed that did not contain prefix."
-
-        log.debug(f"Trigger! {message}")
-
-        # remove command prefix and make lowercase
-        raw_command: str = message.lstrip(cls.prefix)
-
-        words = []
-        words_eol = []
-        remaining = raw_command
-        while True:
-            words_eol.append(remaining)
-            try:
-                word, remaining = remaining.split(maxsplit=1)
-            except ValueError:
-                # we couldn't split -> only one word left
-                words.append(remaining)
+    if words[0] in _registered_commands.keys():
+        cmd = _registered_commands[words[0]]
+    else:
+        for key, value in _rules.items():
+            if key.match(words[0]) is not None:
+                cmd = value
                 break
-            else:
-                words.append(word)
+        else:
+            raise CommandNotFoundException(f"Unable to find command {words[0]}")
 
-        trigger = Trigger.from_bot_user(cls.bot, sender, channel, words, words_eol)
+    return await cmd(bot, trigger)
 
-        # make command case insensitive
-        words[0] = words[0].lower()
-        if words[0] in cls._registered_commands.keys():
-            cmd = cls._registered_commands[words[0]]
-        else:
-            for key, value in cls._rules.items():
-                if key.match(words[0]) is not None:
-                    cmd = value
-                    break
+
+def _register(func, names: list or str) -> bool:
+    """
+    Register a new command
+    :param func: function
+    :param names: names to register
+    :return: success
+    """
+    if isinstance(names, str):
+        names = [names]  # idiot proofing
+
+    # transform commands to lowercase
+    names = [name.lower() for name in names]
+
+    if func is None or not callable(func):
+        # command not callable
+        return False
+
+    else:
+        for alias in names:
+            if alias in _registered_commands:
+                # command already registered
+                raise NameCollisionException(f"attempted to re-register command(s) {alias}")
             else:
-                raise CommandNotFoundException(f"Unable to find command {words[0]}")
+                formed_dict = {alias: func}
+                _registered_commands.update(formed_dict)
```

Comment:
You know while we're here, this could be simplified to `_registered_commands[alias] = func`.